Automattic/newspack-sponsors
# Feat/show hide author categories options

### All Submissions:






### Changes proposed in this Pull Request:



Adds new options for sponsors:

* **Sponsorship Byline Display** - For native sponsors, lets you choose to display both author bylines and sponsor info for posts assigned this sponsor. Setting this option to allow bylines will also display author bios alongside the sponsor info at the bottom of the post.
* **Sponsorship Category Display** - For native sponsors, lets you choose to display both sponsor flags and post categories for posts assigned this sponsor.
* **Underwriter blurb style** - For underwriters, lets you choose whether to display the underwriter blurb with the usual styles or with a simplified plain-text style.
* **Underwriter placement** - For underwriters, lets you choose whether to display the underwriter blurb at the top or bottom of the post.

All new options can also be set on a per-post basis. Posts can also set to force all sponsors to display in native or underwritten style (so that a single sponsor can be used for both native and underwritten posts). If any of these settings are set on the post, these settings override the sponsor's settings for that post.

All options affect singular posts, archive templates, Homepage Posts blocks, and Post Carousel blocks. Requires https://github.com/Automattic/newspack-theme/pull/1833 and https://github.com/Automattic/newspack-blocks/pull/1156 to have any effect.

Closes #71, #175, #176, and #177.

### How to test the changes in this Pull Request:

1. Create or edit a sponsor. Confirm the new options in the Sponsor Settings sidebar panel. Make sure to toggle between native and underwritten scopes to see all new options.
2. Assign the sponsor to a post and test that the new options behave as expected in the singular post template, archive template, Homepage Posts block, and Post Carousel block. For the blocks, confirm both in the editor and on the front-end and with and without the "show author" and "show category" options.
3. Edit the post and confirm that the Sponsor Settings panel now exists here, too. (It should be available for any post type that can be assigned a sponsor.) The default values should be set to "Inherit from sponsor".

<img width="274" alt="Screen Shot 2022-06-03 at 11 29 26 AM" src="https://user-images.githubusercontent.com/2230142/171916050-505bb42b-1bea-487f-b896-6d0b655faa26.png">

4. Test each of the override options and confirm that the singular post shows or hides the relevant info and styles according to the override settings.

### Other information:
